tests/bsim/bluetooth/host/misc/sample_test/src/main.c

```diff
@@ -9,11 +9,35 @@
 #include "bs_tracing.h"
 #include "bstests.h"
 #include "babblekit/testcase.h"
+#include "testlib/log_utils.h"
+
 
 extern void entrypoint_dut(void);
 extern void entrypoint_peer(void);
 extern enum bst_result_t bst_result;
 
+static unsigned long runtime_log_level = LOG_LEVEL_INF;
```

Comment:
This should be a local variable in `test_args`. It is not used outside of it.

Reply:
My apologies, I will create a fresh PR for this with all review comments. My question is `bt_testlib_log_level_set` this function is breaking it so any suggestions regarding this?

Reply:
> @Thalley My apologies, I will create a fresh PR for this with all review comments. 

Please don't create a new PR :) You can squash the commits instead

> My question is bt_testlib_log_level_set this function is breaking it so any suggestions regarding this?

How do you mean?

Reply:
> Please don't create a new PR :) You can squash the commits instead
I will do that, thanks.
> How do you mean?
```

d_00: WARNING: (src/bs_pc_base.c:146): Previous lock file found /tmp/bs_combine/tests_bsim_bluetooth_host_misc_sample_test/2G4.d0.lock
d_00: WARNING: (src/bs_pc_base.c:191): Found previous lock owned by DEAD process (pid was 874088), will attempt to take over
d_01: WARNING: (src/bs_pc_base.c:146): Previous lock file found /tmp/bs_combine/tests_bsim_bluetooth_host_misc_sample_test/2G4.d1.lock
d_01: WARNING: (src/bs_pc_base.c:191): Found previous lock owned by DEAD process (pid was 874089), will attempt to take over
p_2G4: WARNING: (src/bs_pc_base.c:146): Previous lock file found /tmp/bs_combine/tests_bsim_bluetooth_host_misc_sample_test/2G4.phy.lock
p_2G4: WARNING: (src/bs_pc_base.c:191): Found previous lock owned by DEAD process (pid was 874090), will attempt to take over
d_00: @00:00:00.000000 INFO: (WEST_TOPDIR/zephyr/tests/bsim/bluetooth/host/misc/sample_test/src/main.c:29): arg : 4

d_00: @00:00:00.000000  <err> [irq] os: Halting system
d_01: @00:00:00.000000  *** Booting Zephyr OS build v3.7.0-rc3-80-g9f25de06cb55 ***
d_01: @00:00:00.000000 INFO: Test start: peer
p_2G4: WARNING: (src/bs_pc_base.c:408): Device 0 left the party unsuspectingly.. I treat it as if it disconnected
timeout: the monitored command dumped core
/home/combine/zephyrproject/zephyr/tests/bsim/sh_common.source: line 22: 874232 Trace/breakpoint trap   $@
timeout: sending signal TERM to command ‘/home/combine/zephyrproject/zephyr/../tools/bsim/bin/bs_nrf52_bsim_tests_bsim_bluetooth_host_misc_sample_test_prj_conf’
timeout: sending signal TERM to command ‘./bs_2G4_phy_v1’
timeout: sending signal KILL to command ‘/home/combine/zephyrproject/zephyr/../tools/bsim/bin/bs_nrf52_bsim_tests_bsim_bluetooth_host_misc_sample_test_prj_conf’
/home/combine/zephyrproject/zephyr/tests/bsim/sh_common.source: line 22: 874233 Killed                  $@

```

Reply:
You never enable the config for runtime filtering, please read my comment here: https://github.com/zephyrproject-rtos/zephyr/pull/75906#issuecomment-2230036673 and see step 4.

Reply:
Also, please note that the variable cannot be local. It need to be a global because the run time filtering need to happen in the code of the device itself else, the log module will not be registered.

Reply:
> Also, please note that the variable cannot be local. It need to be a global because the run time filtering need to happen in the code of the device itself else, the log module will not be registered.

Not sure I follow this. 
`unsigned long runtime_log_level = LOG_LEVEL_INF;` is just a integer value used to call `bt_testlib_log_level_set` with as a value (not a reference). It is never used or referenced after that. In a similar fashion it should be possible to even do `bt_testlib_log_level_set("dut", 3);` without any variable at all. 

Can you elaborate on why it would need to be a global or `static` variable?

Reply:
Yes, sorry. The thing is that you cannot call `bt_testlib_log_level_set` here. You need to call it in the entrypoint function of the device to be able to filter because the log module is declared there.
The variable need to be global, not static.

Reply:
Ah, so the issue is that `bt_testlib_log_level_set` cannot be called in the `test_args_f` function, but needs to done in the `test_main_f` (or some other function in between?). 

In that case, then yes, the shell argument needs to be stored somewhere in the meantime. 
It may be a moot point though as pointed out by https://github.com/zephyrproject-rtos/zephyr/pull/75906#discussion_r1681068650 there may be a better way of doing this than using the `test_args_f`

Reply:
> Ah, so the issue is that `bt_testlib_log_level_set` cannot be called in the `test_args_f` function, but needs to done in the `test_main_f` (or some other function in between?).

It can be called, it will run fine, but it will have no effect on the log level of "dut" in our case.

> It may be a moot point though as pointed out by [#75906 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/75906#discussion_r1681068650) there may be a better way of doing this than using the `test_args_f`

I don't see how it will solve the present issue, `bt_testlib_log_level_set` will still need to be called in the entrypoint function (or somewhere in the actual test code as you said). I feel like it doesn't change a lot of thing.
IMO we should stick with the first proposal and focus on having a first approach working :slightly_smiling_face:

Reply:
> It can be called, it will run fine, but it will have no effect on the log level of "dut" in our case.

I'm clearly missing some information in order to proper review this PR then. 
The logs are registered as `LOG_MODULE_REGISTER(dut, LOG_LEVEL_DBG);`, so the logging module should already be initialized with `LOG_LEVEL_DBG`, so why cannot we call `bt_testlib_log_level_set` (with effect) here? 

The issue is a bit on clear on this as well: Do we want to set the log level per log module, or all of them? 
If the former, then this needs to be handled in `dut.c` as this PR right now only handles "dut". 
If the latter, then this PR needs to gather all the log modules defined in the test files (only), and apply the log level to all of them. 

> IMO we should stick with the first proposal and focus on having a first approach working

I'd rather we come up with a proper design rather than just winging it :D 

I think there are some fundamental things we need to agree on before going forwards such as

- Whether to apply the log level to just a single logging module or multiple (a single test may define multiple log modules)
- Whether to have individual command line handler per source file (as this PR somewhat does), or handle it more generically in the bt_testlib or the babblekit (as this is not really BT related). 
- Whether to handle it in the test_args_f or as a command line arg as suggested by https://github.com/zephyrproject-rtos/zephyr/pull/75906/files#r1681068650

Reply:
> I'm clearly missing some information in order to proper review this PR then. The logs are registered as `LOG_MODULE_REGISTER(dut, LOG_LEVEL_DBG);`, so the logging module should already be initialized with `LOG_LEVEL_DBG`, so why cannot we call `bt_testlib_log_level_set` (with effect) here?

The log module is registered in Zephyr but when `test_args` is run it's within BabbleSim, the device haven't booted yet.

> Do we want to set the log level per log module, or all of them?

Again, I really think that is a detail. We are in a sample test that can evolve or be updated later. We could do complex stuff like having a pair of argument, one that specify the module and the other that give the log level. But that's more complex and I think it's good to start small ^^